### DISCUSSION TITLE
tacotron2 TTS model trained - what now?

### DISCUSSION THREAD

**Discussion by padmalcom** [speechbrain/speechbrain#1679] [General]
Hi, I trained a German model for 20 epochs and can't wait to hear the results. Unfortunately, the receipt does not tell what I do with my checkpoint. My exact question is, which files do I have to copy into the final folder, to be able to load the model for inference?

ere is my code to test my model which, obviously, does not work (see below). When copying the checkpoint files into the folder model, it askes for a hyperparams.yaml. If I copy the hyperparams.yaml from my training configuration, it misses the pretrainer key in the yaml. Do you have any guide on how to prepare the files?

```
import torchaudio
from speechbrain.pretrained import Tacotron2
from speechbrain.pretrained import HIFIGAN

# Intialize TTS (tacotron2) and Vocoder (HiFIGAN)
tacotron2 = Tacotron2.from_hparams(source="model")
hifi_gan = HIFIGAN.from_hparams(source="speechbrain/tts-hifigan-ljspeech", savedir="tmpdir_vocoder", run_opts={"device":"cuda"})

# Running the TTS
mel_output, mel_length, alignment = tacotron2.encode_text("Mary had a little lamb")

# Running Vocoder (spectrogram-to-waveform)
waveforms = hifi_gan.decode_batch(mel_output)

# Save the waverform
torchaudio.save('example_TTS.wav',waveforms.squeeze(1), 22050)
```

**Comment by BenoitWang** [comment]
Hi @padmalcom, 

You can put your `model.ckpt` and your `hparams.yaml` into the folder and declare a pretrainer class to load the model. Here's an example for [hparams.yaml](https://huggingface.co/speechbrain/tts-tacotron2-ljspeech/blob/main/hyperparams.yaml)

You can check the docstring for the [from_hparams](https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/pretrained/interfaces.py#:~:text=def-,from_hparams) function and a tutorial [here](https://colab.research.google.com/drive/1137MD-gAERblRAVhxDnwNXV3YyAiwtxV?usp=sharing).

**Comment by padmalcom** [comment]
Hi @BenoitWang,
thank you, I took your TTS model on huggingface.co as example and created all the required files. I uploaded my results here: https://huggingface.co/padmalcom/tts-tacotron2-german

P.S. I'm still struggling with loading the model from a local folder (I'm on Windows...) since creating the symlinks in your fetch.py seems not to work here. But loading the model from huggingface works fine. I'm training for three days now, results are decent at epoch 30. I'll keep going for a bit.

**Reply by BenoitWang** [reply]
Hi @padmalcom,

I tested your model and listened to some audios generated and I'm glad that it worked. For the improvements, for now maybe just go more epochs and train a hifi-gan if necessary.

Good luck!

**Comment by padmalcom** [comment]
Hi @BenoitWang,
do you know if a hifi-gan model is language independent? I worked with wavernn and it seemed to work with English and German even if only trained on an English dataset.

**Reply by BenoitWang** [reply]
From what I know the vocoders are weakly dependant to languages, just as you see in your experiments. But they are more dependant to speakers, so training the vocoder may improve the quality.
